server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -403,16 +407,23 @@ private CompletionStage<Result> renderErrorOrRedirectToNextBlock(
     if (featureFlags.isProgramEligibilityConditionsEnabled(request)
         && !roApplicantProgramService.isBlockEligible(blockId)) {
       CiviFormProfile submittingProfile = profileUtils.currentUserProfile(request).orElseThrow();
-      return supplyAsync(
-          () ->
-              ok(
-                  ineligibleBlockView.render(
-                      request,
-                      submittingProfile,
-                      roApplicantProgramService,
-                      applicantName,
-                      messagesApi.preferred(request),
-                      applicantId)));
+      try {
+        String externalLink = programService.getProgramDefinition(programId).externalLink();
```

Comment:
**suggestion:** Shouldn't this only be this link if it's set and if not it should be the program details (here and below)?

Reply:
That part is handled in the UI.. I didn't want to call ProgramService calls in UI hence it is done in controller.